tests/test_gcapi.py

```diff
@@ -5,6 +5,10 @@
 from gcapi import Client, cli
 
 
+def pytest_sessionfinish(session, exitstatus):
+    Client.force_close_open_async_clients()
```

Comment:
> How do we ensure that users do this in their client code?

[atexit](https://docs.python.org/3.6/library/atexit.html)

> Does this need to be added to every test file now? Wouldn't it make more sense to do it once in conftest.py?

I am not super-sure right now that this will be the final version of this test code yet. But in principle, yes. Some httpx cleanup code runs before the atexit code can engage, because that atexit code, by definition runs when python itself starts to shutdown and the test-routines have all long been terminated. Httpx then started to complain that AsyncClients from httpx were not shutdown properly, swamping my console with warnings. I added the test function to tame the beast and be able to work on actual errors.

I do not think this is a brilliant solution yet. I think that for once, either a `__del__` or a weakref-callback might be better choices here. Will see...

Reply:
Yes, but what about `conftest.py`??

Reply:
I want to see if I can remove them entirely first. Otherwise, it sounds like a way better solution, indeed.